lib/handler/connect.c

```diff
@@ -913,23 +923,55 @@ static void on_generator_dispose(void *_self)
 static int on_req(h2o_handler_t *_handler, h2o_req_t *req)
 {
     struct st_connect_handler_t *handler = (void *)_handler;
-    h2o_iovec_t host;
-    uint16_t port;
+    h2o_iovec_t host = {};
+    uint16_t port = 0;
     int is_tcp;
 
     if (h2o_memis(req->input.method.base, req->input.method.len, H2O_STRLIT("CONNECT"))) {
-        is_tcp = 1;
+        if (req->datagram_format == H2O_DATAGRAM_FORMAT_RFC) { //method is CONNECT but parsing of headers found RFC connect-udp
+            is_tcp = 0;
+        } else {
+            is_tcp = 1;
+        }
     } else if (h2o_memis(req->input.method.base, req->input.method.len, H2O_STRLIT("CONNECT-UDP"))) {
         is_tcp = 0;
     } else {
-        return -1;
-    }
-
-    if (h2o_url_parse_hostport(req->authority.base, req->authority.len, &host, &port) == NULL || port == 0 || port == 65535) {
         h2o_send_error_400(req, "Bad Request", "Bad Request", H2O_SEND_ERROR_KEEP_HEADERS);
         return 0;
```

Comment:
That's a keen observation.

I agree that we should preferably return -1 if the request is not a CONNECT or CONNECT-UDP request, as people might simply place a file handler (that handles any GET request) before the connect handler.

Reply:
Yes I see my changes have gotten rid of the -1 return. It will not accept something else but a connect (or connect-udp) but it returns 0 and returns a 400 currently - this should be fixed.